lib/board.rb

```diff
@@ -0,0 +1,60 @@
+ class Board
+
+	attr_accessor :cells
+
+	#can reset the state of the cells in the board
+  #sets the cells of the board to a 9 element array of " "
+	def reset!
+		@cells = Array.new(9, " ")
+	end
+
+	#sets the cells of the board to a 9 element array of " "
+	def initialize(cells=nil)
+		reset!
+	end
+
+	# prints the board
+	def display
+    puts " #{@cells[0]} | #{@cells[1]} | #{@cells[2]} "
+    puts "-----------"
+    puts " #{@cells[3]} | #{@cells[4]} | #{@cells[5]} "
+    puts "-----------"
+    puts " #{@cells[6]} | #{@cells[7]} | #{@cells[8]} "
+  end
+
+  #takes in user input and returns the value of the board cell 
+  def position(input)
+    cells[input.to_i - 1]
+  end
+
+	#returns true for a full board
+	def full?
+  	cells.all?{|token| token == "X" || token == "O"}
+	end
+
+	#returns the amount of turns based on cell value
+  def turn_count  
+		cells.count {|c| c != " "}
+	end
+
+	def taken?(input)
+		#(position(input) == "X" || position(input) == "O") ? true : false
+			cells[input.to_i - 1] == "X" || cells[input.to_i - 1] == "O" ? true : false
```

Comment:
Avoid using the ternary operator to return `true` and `false`. Instead, rely on the implicit boolean return value of the expression. 

`cells[input.to_i - 1] == "X" || cells[input.to_i - 1] == "O"`